### PR TITLE
Turning `if` chains into shorter `switch` statements

### PR DESCRIPTION
**Pull request by nate-thegrate**
This pull request is part of the effort to solve issue #144903.

In the past, my efforts to reduce line length involved refactoring away from switch statements, but unlike [yesterday's PR](https://github.com/flutter/flutter/pull/144905), this one is full of switch statements that make things more concise!

### CODE REVIEW

File: packages/flutter/lib/src/material/navigation_bar.dart [hunk 1]
```diff
@@ -1277,21 +1277,15 @@ class _CurvedAnimationBuilderState extends State<_CurvedAnimationBuilder> {
   // direction.
   void _updateStatus(AnimationStatus status) {
     if (_animationDirection != status) {
-      setState(() {
-        _animationDirection = status;
-      });
+      setState(() => _animationDirection = status);
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/navigation_bar.dart:1280]: In general, we prefer styling `setState` as a block like it was before. Can you revert this formatting change?

Comment (goderbauer) [reply packages/flutter/lib/src/material/navigation_bar.dart:1280]: same below.

**Review by goderbauer — APPROVED** [review]
LGTM with a tiny formatting nit

File: packages/flutter/test/cupertino/nav_bar_transition_test.dart [hunk 2]
```diff
@@ -59,26 +59,24 @@ Future<void> startTransitionBetween(
...
-  if (navBar is CupertinoNavigationBar || navBar == null) {
-    return CupertinoPageScaffold(
-      navigationBar: navBar as CupertinoNavigationBar? ?? const CupertinoNavigationBar(),
-      child: const Placeholder(),
-    );
-  } else if (navBar is CupertinoSliverNavigationBar) {
-    return CupertinoPageScaffold(
-      child: CustomScrollView(
-        slivers: <Widget>[
+  switch (navBar) {
```
Comment (loic-sharma) [inline packages/flutter/test/cupertino/nav_bar_transition_test.dart:62]: Could this be a switch expression?

**Review by loic-sharma — APPROVED** [review]
Thanks for the excellent contribution!

File: packages/flutter/lib/src/material/navigation_bar.dart (hunk 1 above)
Comment (nate-thegrate) [reply packages/flutter/lib/src/material/navigation_bar.dart:1280]: I had a look through the repo and you're right—an overwhelming majority of `setState` calls use a function body even if there's just one statement.

I was surprised at how many were formatted as a single line without the `=>` syntax:

```dart
setState(() { _value = newValue; });
```

I guess now I know to use curly braces in the Flutter repo from now on.

File: packages/flutter/test/cupertino/nav_bar_transition_test.dart (hunk 2 above)
Comment (nate-thegrate) [reply packages/flutter/test/cupertino/nav_bar_transition_test.dart:62]: You're a man of great taste: I'm currently on my [11th PR](https://github.com/flutter/flutter/pull/144580) in an attempt to refactor the entire repo to use switch expressions.

I like having each switch expression case as a single line, but I guess that's just a personal preference.

The real issue here is the default case: throwing an error works fine, but `assert(false); return null;` sadly can't be done.

Comment (loic-sharma) [reply packages/flutter/test/cupertino/nav_bar_transition_test.dart:62]: Ah right, missed that. Resolving!

**Comment by nate-thegrate** [comment]
Gotta love it when "Google testing" decides to flake twice in a row. @loic-sharma would you mind re-applying `autosubmit`?

**Comment by nate-thegrate** [comment]
Thank you!

**Pull request merged**